### ISSUE TITLE
Investigate Google Docs API

### ISSUE THREAD

**Issue by filesync-online** [filesync-online/filesync#7]
We need to see how we can auto dump data into Google sheets.

**Comment by pere5** [comment]
@filesync-online https://developers.google.com/identity/protocols/OAuth2ServiceAccount

Reading there now, but you should read as well as you will have to do a lot of the setup in your g-suite

**Comment by pere5** [comment]
@filesync-online You will have to follow the instructions carefully in the link above, then supply me with service account's email address, and public/private key pair. And possibly API keys.

https://developers.google.com/sheets/api/guides/authorizing

**Comment by filesync-online** [comment]
Service Account ID: filesync@kpi-automation-243414.iam.gserviceaccount.com
User Email: mentormedier@gmail.com

Key: [kpi-automation-243414-1f80fda52d31.zip](https://github.com/filesync-online/filesync/files/3282845/kpi-automation-243414-1f80fda52d31.zip)

![image](https://user-images.githubusercontent.com/50658466/59379617-62c10980-8d57-11e9-9c13-cc4c4a87ae2b.png)

**Comment by filesync-online** [comment]
[Start Stopp KPI.xlsx](https://github.com/filesync-online/filesync/files/3282858/Start.Stopp.KPI.xlsx)
The Google sheets that it will be in!

**Comment by pere5** [comment]
@filesync-online I need the URL of the spread sheet as well.

**Comment by pere5** [comment]
@filesync-online the spreadsheetId

**Comment by filesync-online** [comment]
@pere5 Start Stopp KPI
https://docs.google.com/spreadsheets/d/1kfqbk-93dYJhUNCV75GyY0hGnJkTbTpO2yq6fzxV4B4/edit#gid=0

**Comment by pere5** [comment]
@filesync-online getting this, need you to take action:

- Google Sheets API has not been used in project 643665908546 before or it is disabled. Enable it by visiting https://console.developers.google.com/apis/api/sheets.googleapis.com/overview?project=643665908546 then retry.

- "reason" : "accessNotConfigured"

"""
403 Forbidden { "code" : 403, "errors" : [ { "domain" : "usageLimits", "message" : "Google Sheets API has not been used in project 643665908546 before or it is disabled. Enable it by visiting https://console.developers.google.com/apis/api/sheets.googleapis.com/overview?project=643665908546 then retry. If you enabled this API recently, wait a few minutes for the action to propagate to our systems and retry.", "reason" : "accessNotConfigured", "extendedHelp" : "https://console.developers.google.com" } ], "message" : "Google Sheets API has not been used in project 643665908546 before or it is disabled. Enable it by visiting https://console.developers.google.com/apis/api/sheets.googleapis.com/overview?project=643665908546 then retry. If you enabled this API recently, wait a few minutes for the action to propagate to our systems and retry.", "status" : "PERMISSION_DENIED" }
"""

**Comment by filesync-online** [comment]
@pere5 Fixed!

**Comment by pere5** [comment]
@filesync-online 
I'm not sure if this is the right way but it seems like the way I'm calling you sheet, it needs to be shared with the service account for me to be able to access it:

https://github.com/googleapis/google-api-ruby-client/issues/461
"""
Did you share the sheet with the service account? Service accounts are effectively separate users and need to be on the ACLs the same as any other user.
"""

**Comment by filesync-online** [comment]
@pere5 I thought since you had account wide access you wouldn't need specific access to that sheet. Anyway, I have now shared that specific sheet with your email (filesync@kpi-automation-243414.iam.gserviceaccount.com) as well.

**Comment by pere5** [comment]
@filesync-online Yes I know, I'll sort this out. But it seems to be working this shared way now at least.
